litestar/_openapi/plugin.py

```diff
@@ -43,13 +49,52 @@ def _build_openapi_schema(self) -> OpenAPI:
 
     def provide_openapi(self) -> OpenAPI:
         if not self._openapi_schema:
-            self._openapi_schema = self._build_openapi_schema()
+            self._openapi_schema = self._build_openapi()
         return self._openapi_schema
 
+    @staticmethod
+    def provide_openapi_schema(__openapi: Any) -> Dict[str, Any]:  # noqa: UP006
```

Comment:
The double underscore is to avoid conflicts with any user DI that might be using the `openapi` name.

This causes the dependencies to have to be declared under the name `_OpenAPIPlugin__openapi` due to [name mangling](https://en.wikipedia.org/wiki/Name_mangling), but maybe better to just be explicit about this and name the deps `_openapi_plugin_openapi` or smth.
```python-console
>>> from msgspec import Struct
>>> class MyStruct(Struct):
...   __a: str
... 
>>> MyStruct(__a="abc")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Unexpected keyword argument '__a'
>>> MyStruct(_MyStruct__a="abc")
MyStruct(_MyStruct__a='abc')
```